src/gtk/toga_gtk/window.py

```diff
@@ -106,10 +106,8 @@ def set_content(self, widget):
             child._impl.container = widget
 
     def show(self):
-        self.native.show_all()
-
-        # Now that the content is visible, we can do our initial hinting,
-        # and use that as the basis for setting the minimum window size.
+        # Now we will do our initial hinting, and use that as the basis
+        # for setting the minimum window size.
```

Comment:
> My recollection is that GTK widgets didn't return a meaningful size until they are visible

Yes this is correct, exactly when it is realize. However with this changes or without it there is no a returned meaningful size at all because `Gtk.Fixed` wait us to set these sizes. For other widgets, when we add it to the container we call `show_all()` in `base.py` file so, as I understand, we get the meaningful sizes for these widgets.